src/Microsoft.AspNetCore.SystemWebAdapters/Internal/NonGenericCollectionWrapper.cs

```diff
@@ -20,7 +20,7 @@ public NonGenericCollectionWrapper(ICollection<T> collection)
 
         public bool IsSynchronized => false;
 
-        public object SyncRoot => null!;
+        public object SyncRoot => _collection;
```

Comment:
I think this is very unlikely to matter one way or another, but I looked at how we've done this in the framework before and there appears to be some precedent for just creating a new syncroot object if the wrapped collection doesn't have one: https://referencesource.microsoft.com/#mscorlib/system/collections/objectmodel/readonlydictionary.cs,434

Reply:
I've updated to use that pattern (although we don't need it to be thread-safe)